plugins/janus_audiobridge.c

```diff
@@ -7852,6 +7881,47 @@ static void *janus_audiobridge_mixer_thread(void *data) {
 					}
 				}
 			}
+
+#if HAVE_LIBOGG
+			for(GList *annc = anncs_list; annc; annc = annc->next) {
+				janus_audiobridge_participant *anncp = (janus_audiobridge_participant *)annc->data;
+
+				if(anncp->annc == NULL || g_atomic_int_get(&anncp->destroyed))
+					continue;
+
+				if(!anncp->annc->to && !anncp->annc->exclude)
+					continue; // already mixed
+
+				GList *first = g_list_first(anncp->inbuf);
+				janus_audiobridge_rtp_relay_packet *pkt = (janus_audiobridge_rtp_relay_packet *)(first ? first->data : NULL);
+				curBuffer = (opus_int16 *)((pkt && pkt->length && !pkt->silence) ? pkt->data : NULL);
+				if(!curBuffer)
+					continue;
+
+				const char* to = anncp->annc->to;
+				const char* exclude = anncp->annc->exclude;
+				if(to) {
+					if(0 == g_strcmp0(p->display, to)) {
```

Comment:
So `to` and `exclude` are display names? I think this is very fragile. Display names are _not_ unique, and can be changed dynamically by people in the room. If you want this to be reliable and consistent, it should be user IDs instead (which means making the same checks we do on whether they're numbers or string IDs).

Reply:
Good point. I need to think if it suitable in our case. Thanks!